session/session.go

```diff
@@ -851,6 +854,69 @@ func (s *session) SetGlobalSysVar(name, value string) error {
 	return errors.Trace(err)
 }
 
+//AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (s *session) AddGlobalBind(originSQL string, bindSQL string, defaultDB string, charset string, collation string) (err error) {
+	ctx := context.Background()
+
+	_, err = s.Execute(ctx, "BEGIN")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() {
+		if err == nil {
+			_, err = s.execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := s.execute(ctx, "ROLLBACK")
+			terror.Log(errors.Trace(rbErr))
+		}
+	}()
+
+	originSQL = getEscapeCharacter(originSQL)
+	sql := fmt.Sprintf("SELECT status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
```

Comment:
So there is no available index for this statement?

Reply:
Add global bind is not used often, so we don't create a index for (original_sql,default_db)

Reply:
It is better to not assume it. And even if it is, there may still lot of binds which will lead to bad performance and affect other statements.

Reply:
ok